crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -244,9 +245,25 @@ impl SearchPaths {
             static_paths.push(SearchPath::extra(system, path)?);
         }
 
+        // Most search paths come from user configuration, so we should just trust that the user
+        // has specified their search paths in the correct order -- if not, that's on the user.
+        // First-party search paths can be implicitly inferred by ty, however (`./src` is a common case),
+        // so for first-party search paths it makes sense to sort them into a sensible order.
+        // We make sure here that child directories always take precedence over parent directories.
```

Comment:
Should we move this to where we add the default search paths. I feel like we shouldn't apply this sorting when it's a user-provided list.